### ISSUE TITLE
Console Build fails to update resources in cloud

### ISSUE THREAD

**Issue by walter-scott** [aws-amplify/amplify-console#71]
**Describe the bug**
The Amplify Console fails to builds my **dev** environent front end because it fails in building the backend (specifically the GraphQL API connected databases). It fails because it is trying to create resources that already exist. 

**To Reproduce**
I don't understand why the problem is happening. However I can tell you what I was doing before it happened: I develop my application locally, on the dev branch and dev env. Then I use the amplify cli to push and then I commit my changes to the dev branch. Then I push the changes to GitHub which triggers the amplify console. I then merge my dev branch to the master branch (which the amplify console knows is connected to my prod env), use amplify push, commit the changes and then push to GitHub. 

**Expected behavior**
I expect both  branches build and deploy the front end through the console and not get stopped by building the backend that I have already pushed through the cli.

**Additional context**
1. I can run amplify push from the amplify cli and it complies my graphql schema and builds successfully.

1.  This only happens for my dev environment, my prod environment is building fine.

1. Well, I think it is building fine. For the prod env,  instead of compiling my schema in the console build, it says `Could not read cloudformation template at path: /codebuild/output/src022018288/src/intro-web-app/amplify/backend/api/IntroWebAppAPI/build/cloudformation-template.json`.  Of which the dev env was saying until it started trying to build resources.

**Sample code**
Beginning of last build for dev (fail): 
```html
                                 # Starting phase: build
                                 # Executing command: amplifyPush --simple
2019-08-22T14:19:27.834Z [INFO]: # Getting Amplify CLI Cloud-Formation stack info from environment cache
2019-08-22T14:19:27.844Z [INFO]: # Start initializing Amplify environment: dev
2019-08-22T14:19:27.845Z [INFO]: STACKINFO={
                                 "awscloudformation": {
                                 "AuthRoleName": "",
                                 "UnauthRoleArn": "",
                                 "AuthRoleArn": "",
                                 "Region": "",
                                 "DeploymentBucketName": "",
                                 "UnauthRoleName": ",
                                 "StackName": "",
                                 "StackId": ""
                                 }
                                 }
                                 # Importing Amplify environment: dev (amplify env import)
2019-08-22T14:19:39.380Z [INFO]: Successfully added environment from your project
2019-08-22T14:19:39.441Z [INFO]: # Initializing existing Amplify environment: dev (amplify init)
2019-08-22T14:19:43.508Z [INFO]: Note: It is recommended to run this command from the root of your app directory
2019-08-22T14:19:46.741Z [WARNING]: - Initializing your environment: dev
2019-08-22T14:19:47.132Z [WARNING]: ✔ Initialized provider successfully.
2019-08-22T14:19:47.737Z [INFO]: GraphQL schema compiled successfully.
                                 Edit your schema at /codebuild/output/src257450130/src/intro-web-app/amplify/backend/api/IntroWebAppAPI/schema.graphql or place .graphql files in a directory at /codebuild/output/src257450130/src/intro-web-app/amplify/backend/api/IntroWebAppAPI/schema
2019-08-22T14:20:03.428Z [WARNING]: - Updating resources in the cloud. This may take a few minutes...
2019-08-22T14:20:08.987Z [INFO]: 
2019-08-22T14:20:09.002Z [INFO]: UPDATE_IN_PROGRESS intro-web-app-dev-20190703083122 AWS::CloudFormation::Stack Thu Aug 22 2019 14:20:03 GMT+0000 (Coordinated Universal Time) User Initiated
2019-08-22T14:20:13.982Z [INFO]: 
2019-08-22T14:20:13.984Z [INFO]: UPDATE_IN_PROGRESS storageIntroWebAppStorage AWS::CloudFormation::Stack Thu Aug 22 2019 14:20:10 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS AuthRole                  AWS::IAM::Role             Thu Aug 22 2019 14:20:10 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS UnauthRole                AWS::IAM::Role             Thu Aug 22 2019 14:20:10 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    storageIntroWebAppStorage AWS::CloudFormation::Stack Thu Aug 22 2019 14:20:11 GMT+0000 (Coordinated Universal Time)
2019-08-22T14:20:23.924Z [INFO]: 
2019-08-22T14:20:23.927Z [INFO]: UPDATE_COMPLETE    UnauthRole                     AWS::IAM::Role             Thu Aug 22 2019 14:20:19 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    AuthRole                       AWS::IAM::Role             Thu Aug 22 2019 14:20:19 GMT+0000 (Coordinated Universal Time)
                                 CREATE_IN_PROGRESS UpdateRolesWithIDPFunctionRole AWS::IAM::Role             Thu Aug 22 2019 14:20:22 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS authintrowebapp02c325c6        AWS::CloudFormation::Stack Thu Aug 22 2019 14:20:22 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS analyticsIntroWebApp           AWS::CloudFormation::Stack Thu Aug 22 2019 14:20:22 GMT+0000 (Coordinated Universal Time)
                                 CREATE_IN_PROGRESS UpdateRolesWithIDPFunctionRole AWS::IAM::Role             Thu Aug 22 2019 14:20:23 GMT+0000 (Coordinated Universal Time) Resource creation Initiated
                                 UPDATE_COMPLETE    authintrowebapp02c325c6        AWS::CloudFormation::Stack Thu Aug 22 2019 14:20:23 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_COMPLETE    analyticsIntroWebApp           AWS::CloudFormation::Stack Thu Aug 22 2019 14:20:23 GMT+0000 (Coordinated Universal Time)
2019-08-22T14:20:28.931Z [INFO]: 
2019-08-22T14:20:28.933Z [INFO]: UPDATE_IN_PROGRESS apiIntroWebAppAPI AWS::CloudFormation::Stack Thu Aug 22 2019 14:20:27 GMT+0000 (Coordinated Universal Time)
2019-08-22T14:20:29.024Z [INFO]: 
2019-08-22T14:20:29.025Z [INFO]: UPDATE_IN_PROGRESS intro-web-app-dev-20190703083122-apiIntroWebAppAPI-S3ZZ4VOHLD7T AWS::CloudFormation::Stack Thu Aug 22 2019 14:20:28 GMT+0000 (Coordinated Universal Time) User Initiated
2019-08-22T14:20:39.333Z [INFO]: 
2019-08-22T14:20:39.334Z [INFO]: UPDATE_IN_PROGRESS GraphQLSchema AWS::AppSync::GraphQLSchema Thu Aug 22 2019 14:20:39 GMT+0000 (Coordinated Universal Time)
2019-08-22T14:20:43.933Z [INFO]: 
2019-08-22T14:20:43.935Z [INFO]: CREATE_COMPLETE UpdateRolesWithIDPFunctionRole AWS::IAM::Role Thu Aug 22 2019 14:20:42 GMT+0000 (Coordinated Universal Time)
2019-08-22T14:20:48.933Z [INFO]: 
2019-08-22T14:20:48.935Z [INFO]: CREATE_IN_PROGRESS UpdateRolesWithIDPFunction AWS::Lambda::Function Thu Aug 22 2019 14:20:45 GMT+0000 (Coordinated Universal Time)
                                 CREATE_IN_PROGRESS UpdateRolesWithIDPFunction AWS::Lambda::Function Thu Aug 22 2019 14:20:45 GMT+0000 (Coordinated Universal Time) Resource creation Initiated
                                 CREATE_COMPLETE    UpdateRolesWithIDPFunction AWS::Lambda::Function Thu Aug 22 2019 14:20:45 GMT+0000 (Coordinated Universal Time)
2019-08-22T14:20:53.941Z [INFO]: 
2019-08-22T14:20:53.942Z [INFO]: CREATE_IN_PROGRESS UpdateRolesWithIDPFunctionOutputs Custom::LambdaCallout Thu Aug 22 2019 14:20:49 GMT+0000 (Coordinated Universal Time)
                                 CREATE_IN_PROGRESS UpdateRolesWithIDPFunctionOutputs Custom::LambdaCallout Thu Aug 22 2019 14:20:53 GMT+0000 (Coordinated Universal Time) Resource creation Initiated
                                 CREATE_COMPLETE    UpdateRolesWithIDPFunctionOutputs Custom::LambdaCallout Thu Aug 22 2019 14:20:53 GMT+0000 (Coordinated Universal Time)
2019-08-22T14:20:54.132Z [INFO]: 
2019-08-22T14:20:54.133Z [INFO]: UPDATE_COMPLETE GraphQLSchema AWS::AppSync::GraphQLSchema Thu Aug 22 2019 14:20:53 GMT+0000 (Coordinated Universal Time)
2019-08-22T14:21:04.137Z [INFO]: 
2019-08-22T14:21:04.139Z [INFO]: CREATE_IN_PROGRESS CoursePracticalExam AWS::CloudFormation::Stack Thu Aug 22 2019 14:20:59 GMT+0000 (Coordinated Universal Time)
                                 UPDATE_IN_PROGRESS Course              AWS::CloudFormation::Stack Thu Aug 22 2019 14:20:59 GMT+0000 (Coordinated Universal Time)
                                 CREATE_IN_PROGRESS CoursePracticalExam AWS::CloudFormation::Stack Thu Aug 22 2019 14:21:00 GMT+0000 (Coordinated Universal Time) Resource creation Initiated
                                 UPDATE_IN_PROGRESS ExerciseResult      AWS::CloudFormation::Stack Thu Aug 22 2019 14:21:00 GMT+0000 (Coordinated Universal Time)
2019-08-22T14:21:04.214Z [INFO]: 
2019-08-22T14:21:04.215Z [INFO]: UPDATE_IN_PROGRESS intro-web-app-dev-20190703083122-apiIntroWebAppAPI-S3ZZ4VOHLD7T-ExerciseResult-L3T8N4XAJBYT AWS::CloudFormation::Stack Thu Aug 22 2019 14:21:00 GMT+0000 (Coordinated Universal Time) User Initiated
2019-08-22T14:21:04.273Z [INFO]: 
2019-08-22T14:21:04.274Z [INFO]: CREATE_IN_PROGRESS intro-web-app-dev-20190703083122-apiIntroWebAppAPI-S3ZZ4VOHLD7T-CoursePracticalExam-1BN5KTSLYBDTY AWS::CloudFormation::Stack Thu Aug 22 2019 14:21:00 GMT+0000 (Coordinated Universal Time) User Initiated
                                 CREATE_FAILED      intro-web-app-dev-20190703083122-apiIntroWebAppAPI-S3ZZ4VOHLD7T-CoursePracticalExam-1BN5KTSLYBDTY AWS::CloudFormation::Stack Thu Aug 22 2019 14:21:02 GMT+0000 (Coordinated Universal Time) Export with name e7xjaxrejrh7rnyxanzz72hqta:GetAtt:CoursePracticalExamDataSource:Name is already exported by stack intro-web-app-dev-20190703083122-apiIntroWebAppAPI-S3ZZ4VOHLD7T-PracticalExam-1FVSL1CZZ1Y0I

```

beginning of last build for prod (success): 
```html
                                 # Starting phase: build
                                 # Executing command: amplifyPush --simple
2019-08-21T23:03:59.189Z [INFO]: # Getting Amplify CLI Cloud-Formation stack info from environment cache
2019-08-21T23:03:59.218Z [INFO]: # Start initializing Amplify environment: prod
                                 STACKINFO={
                                 "awscloudformation": {
                                 "AuthRoleName": "",
                                 "UnauthRoleArn": "",
                                 "AuthRoleArn": "",
                                 "Region": "",
                                 "DeploymentBucketName": "",
                                 "UnauthRoleName": "",
                                 "StackName": "",
                                 "StackId": ""
                                 },
                                 "categories": {
                                 "auth": {
                                 "introwebapp02c325c6": {}
                                 }
                                 }
                                 }
                                 # Importing Amplify environment: prod (amplify env import)
2019-08-21T23:04:07.839Z [INFO]: Successfully added environment from your project
2019-08-21T23:04:07.899Z [INFO]: # Initializing existing Amplify environment: prod (amplify init)
2019-08-21T23:04:11.963Z [INFO]: Note: It is recommended to run this command from the root of your app directory
2019-08-21T23:04:14.706Z [WARNING]: - Initializing your environment: prod
2019-08-21T23:04:15.061Z [WARNING]: ✔ Initialized provider successfully.
2019-08-21T23:04:15.110Z [INFO]: Could not read cloudformation template at path: /codebuild/output/src022018288/src/intro-web-app/amplify/backend/api/IntroWebAppAPI/build/cloudformation-template.json
2019-08-21T23:04:15.110Z [WARNING]: - Updating resources in the cloud. This may take a few minutes...
2019-08-21T23:04:15.245Z [WARNING]: ✔ All resources are updated in the cloud
```

**Comment by swaminator** [comment]
@walter-scott we've had similar reports here: https://github.com/aws-amplify/amplify-console/issues/10#issuecomment-511240983. Thanks for providing all that context. We're going to look into this and get back to you.

**Comment by nimacks** [comment]
@walter-scott , the Amplify CLI v2.0.0 is out, can you give that a shot?

**Comment by walter-scott** [comment]
@nimacks I upgraded to Amplify CLI v2.0.0 locally and all went well when I did a simple push of just adding a space to my GraphQL schema. However, the console is doing the same thing - trying to create my Many to Many relationship tables and failing because they already exist. I made sure to put `latest` as the live update and inside provision the console said it switched from v1.12.0 to v2.0.0. Also, the new @auth warning from v2.0.0 is present in the build, so it was definitely running v2.0.0.

Before when I got the error from v1.12.0 it would happen both on the CLI and in the console, so this is weird. 

Is there anything else I should try?
